components/org.wso2.carbon.identity.application.authenticator.apple/src/main/java/org/wso2/carbon/identity/application/authenticator/apple/AppleAuthenticator.java

```diff
@@ -636,6 +645,56 @@ protected String getComponentId() {
         return AppleAuthenticatorConstants.LogConstants.OUTBOUND_AUTH_APPLE_SERVICE;
     }
 
+    /**
+     * Get the i18n key defined to represent the authenticator name.
+     *
+     * @return the 118n key.
+     */
+    @Override
+    public String getI18nKey() {
+
+        return AUTHENTICATOR_APPLE;
+    }
+
+    /**
+     * This method is responsible for validating whether the authenticator is supported for API Based Authentication.
+     *
+     * @return true if the authenticator is supported for API Based Authentication.
+     */
+    @Override
+    public boolean isAPIBasedAuthenticationSupported() {
+
+        return true;
+    }
+
+    /**
+     * This method is responsible to return the scopes as space separated string.
+     *
+     * @param authenticatorProperties   Authenticator properties.
+     * @return Scopes string
+     */
+    @Override
+    protected String getScope(Map<String, String> authenticatorProperties) {
+
+        String scopes = authenticatorProperties.get(IdentityApplicationConstants.Authenticator.OIDC.SCOPES);
+        return scopes.replace(",", " ").replace("+", " ");
```

Comment:
Lets add a null check here

Reply:
also lets take the separators to constants

Reply:
This was addressed in both https://github.com/wso2-extensions/identity-outbound-auth-apple/pull/9/commits/1a9f83d2deac9d435d73d8dcd3e3f08d792db3bc and https://github.com/wso2-extensions/identity-outbound-auth-apple/pull/9/commits/84cbc2f539b418033c4f1c3e4549a75862b4c3f2